tests/testlib/s2n_ktls_test_utils.c

```diff
@@ -138,15 +139,31 @@ ssize_t s2n_test_ktls_recvmsg_io_stuffer(void *io_context, struct msghdr *msg)
         if (no_more_records) {
             break;
         }
-        bool next_record_different_type = next_record_type != *record_type;
+
+        bool next_record_different_type = next_record_type != record_type;
         if (next_record_different_type) {
             break;
         }
+
+        /* There is another record of the same record_type so consume the tag portion of
+         * the header and continue to read.
+         */
+        POSIX_GUARD(s2n_stuffer_skip_read(&io_ctx->ancillary_buffer, S2N_TEST_KTLS_MOCK_HEADER_TAG_SIZE));
     }
 
     return bytes_read;
 }
 
+ssize_t s2n_test_ktls_sendmsg_fail(void *io_context, const struct msghdr *msg)
+{
+    struct s2n_test_ktls_io_fail *io_ctx = (struct s2n_test_ktls_io_fail *) io_context;
+    POSIX_ENSURE_REF(io_ctx);
+    io_ctx->invoked_count++;
+
+    errno = EINVAL;
+    return -1;
+}
```

Comment:
If you want to make this more general, I would suggest adding the errno value to the context. Otherwise, it's specific enough it probably shouldn't be in test_utils.

You can also start with it in a specific test suite and only move it to test_utils if multiple test suites need it.

Reply:
interestingly its not possible to test the blocking behavior fully via this fail imp. but I say we keep the suffer based test since it aligns better with the current s2n_send one and the stuffer better mimics the network